docs/docs/gatsby-core-philosophy.md

```diff
@@ -2,7 +2,177 @@
 title: Gatsby core philosophy
 ---
 
-This is a stub. Help our community expand it.
+Gatsby's core philosophy can be divided into three parts.
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+**First, our vision.** Our vision is to _create better building blocks for the web_, and _build a cohesive "content mesh system"_, in order to _make building websites fun_.
+
+**Second, our tooling philosophy.** In order to make website building fun, Gatsby's tooling must embody certain qualities. These include:
+
+- making the right thing the easy thing
+- progressively disclosure of complexity
+
+**Third, our community philosophy.** We can't do this alone, so we're striving to build the most inclusive community on the web. For that reason, we _work in the open_. At all times, we believe, and strive to communicate, that _you belong here_.
+
+## The Gatsby Vision
+
+### Create new, higher-level web building blocks
+
+Today's building blocks for the web are HTML, CSS, and Javascript; `<img>` and `<a href>` and so on.
+
+We believe that in 20 or 25 years, we'll look back at many of these blocks like we look back at machine code or assembly language today; low-level languages that are great compile targets for higher-level languages that are easier to write in.
+
+Using abstractions like React components, `gatsby-image`, and `gatsby-link`, we've begun to craft this higher-level language. But we're just getting started. Gatsby is a playground for discovering new building blocks for the web.
+
+To [quote Alan Kay](https://www.youtube.com/watch?v=NdSD07U5uBs):
+
+> “You get simplicity by finding slightly more sophisticated building blocks“.
+
+[https://www.gatsbyjs.org/blog/2017-10-16-making-website-building-fun/](Read more here).
+
+### Build a cohesive "content mesh system"
+
+Over the last few years, teams have begun moving from monolithic, integrated CMS setups to a modular "content mesh". Their key challenge: integrating all these pieces together.
+
+Gatsby's goal is to build a "content mesh system" that knits modular, best-of-breed tools in content modelling, authentication, search, analytics, payments, UI frameworks, and so on -- into a unified, high-quality whole.
+
+[Read more here](https://www.gatsbyjs.org/blog/2018-10-04-journey-to-the-content-mesh).
+
+### Make building websites fun
+
+Gatsby's vision is to make website development fun by making it simple.
+
+Fred Brooks wrote in _The Mythical Man-Month_:
+
+> The programmer, like the poet, works only slightly removed from pure thought-stuff. He builds his castles in the air, from air, creating by exertion of the imagination. Few media of creation are so flexible, so easy to polish and rework, so readily capable of realizing grand conceptual structures…
```

Comment:
I know this is a quote, but can we make it more gender-inclusive by paraphrasing with "they" instead of "he"?

Reply:
Good point! can do